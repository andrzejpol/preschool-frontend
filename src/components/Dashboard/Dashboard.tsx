import UserCard from "../UserCard";
import CountChart from "../CountChart";
import AttendanceChart from "../AttendanceChart";
import EventCalendar from "../EventCalendar";

const Dashboard = () => {

  return (
    <div className="p-4 flex gap-4 flex-col md:flex-row">
      <div className="w-full lg:w-2/3 flex flex-col gap-8">
        <div className="flex gap-4 justify-between flex-wrap">
          <UserCard type="student"/>
          <UserCard type="parent"/>
          <UserCard type="teacher"/>
        </div>
        <div className="flex gap-4 flex-col lg:flex-row">
          <div className="w-full lg:w-1/3 h-[450px]">
            <CountChart/>
          </div>
          <div className="w-full lg:w-2/3 h-[450px]">
            <AttendanceChart/>
          </div>
        </div>
      </div>
      <div className="w-full lg:w-1/3 flex flex-col gap-8">
        <EventCalendar/>
      </div>
    </div>
  )
};

export default Dashboard;