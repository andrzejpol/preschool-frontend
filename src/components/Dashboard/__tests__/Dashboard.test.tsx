import React from "react";
import {render} from "@testing-library/react";
import Dashboard from "../Dashboard";

vi.mock("../UserCard", () => ({
  default: ({type}) => <div>Mocked UserCard: {type}</div>,
}));

vi.mock("../CountChart", () => ({
  default: () => <div>Mocked CountChart</div>,
}));

vi.mock("../AttendanceChart", () => ({
  default: () => <div>Mocked AttendanceChart</div>,
}));

vi.mock("../EventCalendar", () => ({
  default: () => <div>Mocked EventCalendar</div>,
}));

describe("Dashboard Component", () => {
  it("matches the snapshot", () => {
    const {asFragment} = render(<Dashboard/>);

    // Snapshot assertion
    expect(asFragment()).toMatchSnapshot();
  });
});
