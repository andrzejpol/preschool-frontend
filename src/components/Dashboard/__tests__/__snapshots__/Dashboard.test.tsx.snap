// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dashboard Component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="p-4 flex gap-4 flex-col md:flex-row"
  >
    <div
      class="w-full lg:w-2/3 flex flex-col gap-8"
    >
      <div
        class="flex gap-4 justify-between flex-wrap"
      >
        <div
          class="rounded-2xl odd:bg-preschoolPrimaryLight even:bg-preschoolSecondaryLight p-4 flex-1 min-w-[130px]"
        >
          <div
            class="flex justify-between items-center"
          >
            <span
              class="text-[10px] bg-white px-2 py-1 rounded-full text-gray-500"
            >
              2024/25
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-info flex items-center justify-center w-[10px] h-[10px] text-gray-500"
              data-icon="info"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 192 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M48 80a48 48 0 1 1 96 0A48 48 0 1 1 48 80zM0 224c0-17.7 14.3-32 32-32l64 0c17.7 0 32 14.3 32 32l0 224 32 0c17.7 0 32 14.3 32 32s-14.3 32-32 32L32 512c-17.7 0-32-14.3-32-32s14.3-32 32-32l32 0 0-192-32 0c-17.7 0-32-14.3-32-32z"
                fill="currentColor"
              />
            </svg>
          </div>
          <h1
            class="text-2xl font-semibold my-4"
          >
            1,234
          </h1>
          <h2
            class="capitalize text-sm font-medium text-gray-500"
          >
            student
          </h2>
        </div>
        <div
          class="rounded-2xl odd:bg-preschoolPrimaryLight even:bg-preschoolSecondaryLight p-4 flex-1 min-w-[130px]"
        >
          <div
            class="flex justify-between items-center"
          >
            <span
              class="text-[10px] bg-white px-2 py-1 rounded-full text-gray-500"
            >
              2024/25
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-info flex items-center justify-center w-[10px] h-[10px] text-gray-500"
              data-icon="info"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 192 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M48 80a48 48 0 1 1 96 0A48 48 0 1 1 48 80zM0 224c0-17.7 14.3-32 32-32l64 0c17.7 0 32 14.3 32 32l0 224 32 0c17.7 0 32 14.3 32 32s-14.3 32-32 32L32 512c-17.7 0-32-14.3-32-32s14.3-32 32-32l32 0 0-192-32 0c-17.7 0-32-14.3-32-32z"
                fill="currentColor"
              />
            </svg>
          </div>
          <h1
            class="text-2xl font-semibold my-4"
          >
            1,234
          </h1>
          <h2
            class="capitalize text-sm font-medium text-gray-500"
          >
            parent
          </h2>
        </div>
        <div
          class="rounded-2xl odd:bg-preschoolPrimaryLight even:bg-preschoolSecondaryLight p-4 flex-1 min-w-[130px]"
        >
          <div
            class="flex justify-between items-center"
          >
            <span
              class="text-[10px] bg-white px-2 py-1 rounded-full text-gray-500"
            >
              2024/25
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-info flex items-center justify-center w-[10px] h-[10px] text-gray-500"
              data-icon="info"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 192 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M48 80a48 48 0 1 1 96 0A48 48 0 1 1 48 80zM0 224c0-17.7 14.3-32 32-32l64 0c17.7 0 32 14.3 32 32l0 224 32 0c17.7 0 32 14.3 32 32s-14.3 32-32 32L32 512c-17.7 0-32-14.3-32-32s14.3-32 32-32l32 0 0-192-32 0c-17.7 0-32-14.3-32-32z"
                fill="currentColor"
              />
            </svg>
          </div>
          <h1
            class="text-2xl font-semibold my-4"
          >
            1,234
          </h1>
          <h2
            class="capitalize text-sm font-medium text-gray-500"
          >
            teacher
          </h2>
        </div>
      </div>
      <div
        class="flex gap-4 flex-col lg:flex-row"
      >
        <div
          class="w-full lg:w-1/3 h-[450px]"
        >
          <div
            class="bg-white rounded-xl w-full h-full p-4"
          >
            <div
              class="flex justify-between items-center"
            >
              <h1
                class="text-lg font-semibold"
              >
                Students
              </h1>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-graduation-cap w-[20px] h-[20px]"
                data-icon="graduation-cap"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 640 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M320 32c-8.1 0-16.1 1.4-23.7 4.1L15.8 137.4C6.3 140.9 0 149.9 0 160s6.3 19.1 15.8 22.6l57.9 20.9C57.3 229.3 48 259.8 48 291.9l0 28.1c0 28.4-10.8 57.7-22.3 80.8c-6.5 13-13.9 25.8-22.5 37.6C0 442.7-.9 448.3 .9 453.4s6 8.9 11.2 10.2l64 16c4.2 1.1 8.7 .3 12.4-2s6.3-6.1 7.1-10.4c8.6-42.8 4.3-81.2-2.1-108.7C90.3 344.3 86 329.8 80 316.5l0-24.6c0-30.2 10.2-58.7 27.9-81.5c12.9-15.5 29.6-28 49.2-35.7l157-61.7c8.2-3.2 17.5 .8 20.7 9s-.8 17.5-9 20.7l-157 61.7c-12.4 4.9-23.3 12.4-32.2 21.6l159.6 57.6c7.6 2.7 15.6 4.1 23.7 4.1s16.1-1.4 23.7-4.1L624.2 182.6c9.5-3.4 15.8-12.5 15.8-22.6s-6.3-19.1-15.8-22.6L343.7 36.1C336.1 33.4 328.1 32 320 32zM128 408c0 35.3 86 72 192 72s192-36.7 192-72L496.7 262.6 354.5 314c-11.1 4-22.8 6-34.5 6s-23.5-2-34.5-6L143.3 262.6 128 408z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <div
              class="w-full h-[75%] relative"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-children w-[50px] h-[50px] absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2"
                data-icon="children"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 640 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M160 0a64 64 0 1 1 0 128A64 64 0 1 1 160 0zM88 480l0-80-17.8 0c-10.9 0-18.6-10.7-15.2-21.1l31.1-93.4L57.5 323.3c-10.7 14.1-30.8 16.8-44.8 6.2s-16.8-30.7-6.2-44.8L65.4 207c22.4-29.6 57.5-47 94.6-47s72.2 17.4 94.6 47l58.9 77.7c10.7 14.1 7.9 34.2-6.2 44.8s-34.2 7.9-44.8-6.2l-28.6-37.8L265 378.9c3.5 10.4-4.3 21.1-15.2 21.1L232 400l0 80c0 17.7-14.3 32-32 32s-32-14.3-32-32l0-80-16 0 0 80c0 17.7-14.3 32-32 32s-32-14.3-32-32zM480 0a64 64 0 1 1 0 128A64 64 0 1 1 480 0zm-8 384l0 96c0 17.7-14.3 32-32 32s-32-14.3-32-32l0-179.5L395.1 321c-9.4 15-29.2 19.4-44.1 10s-19.4-29.2-10-44.1l51.7-82.1c17.6-27.9 48.3-44.9 81.2-44.9l12.3 0c33 0 63.7 16.9 81.2 44.9L619.1 287c9.4 15 4.9 34.7-10 44.1s-34.7 4.9-44.1-10L552 300.5 552 480c0 17.7-14.3 32-32 32s-32-14.3-32-32l0-96-16 0z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <div
              class="flex justify-center gap-16"
            >
              <div
                class="flex flex-col gap-1"
              >
                <div
                  class="w-5 h-5 bg-preschoolPrimaryLight rounded-full"
                />
                <h1
                  class="font-bold"
                >
                  1,234
                </h1>
                <h2
                  class="text-sm text-gray-300"
                >
                  Boys (55%)
                </h2>
              </div>
              <div
                class="flex flex-col gap-1"
              >
                <div
                  class="w-5 h-5 bg-preschoolSecondaryLight rounded-full"
                />
                <h1
                  class="font-bold"
                >
                  1,234
                </h1>
                <h2
                  class="text-sm text-gray-300"
                >
                  Girls (45%)
                </h2>
              </div>
            </div>
          </div>
        </div>
        <div
          class="w-full lg:w-2/3 h-[450px]"
        >
          <div
            class="bg-white rounded-lg p-4 h-full"
          >
            <div
              class="flex justify-between items-center"
            >
              <h1
                class="text-lg font-semibold"
              >
                Students
              </h1>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-clipboard-user w-[20px] h-[20px]"
                data-icon="clipboard-user"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M192 0c-41.8 0-77.4 26.7-90.5 64L64 64C28.7 64 0 92.7 0 128L0 448c0 35.3 28.7 64 64 64l256 0c35.3 0 64-28.7 64-64l0-320c0-35.3-28.7-64-64-64l-37.5 0C269.4 26.7 233.8 0 192 0zm0 64a32 32 0 1 1 0 64 32 32 0 1 1 0-64zM128 256a64 64 0 1 1 128 0 64 64 0 1 1 -128 0zM80 432c0-44.2 35.8-80 80-80l64 0c44.2 0 80 35.8 80 80c0 8.8-7.2 16-16 16L96 448c-8.8 0-16-7.2-16-16z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="w-full lg:w-1/3 flex flex-col gap-8"
    >
      <div
        class="bg-white p-4 rounded-md"
      >
        <div
          class="react-calendar"
        >
          <div
            class="react-calendar__navigation"
          >
            <button
              aria-label=""
              class="react-calendar__navigation__arrow react-calendar__navigation__prev2-button"
              type="button"
            >
              «
            </button>
            <button
              aria-label=""
              class="react-calendar__navigation__arrow react-calendar__navigation__prev-button"
              type="button"
            >
              ‹
            </button>
            <button
              aria-label=""
              class="react-calendar__navigation__label"
              style="flex-grow: 1;"
              type="button"
            >
              <span
                class="react-calendar__navigation__label__labelText react-calendar__navigation__label__labelText--from"
              >
                January 2025
              </span>
            </button>
            <button
              aria-label=""
              class="react-calendar__navigation__arrow react-calendar__navigation__next-button"
              type="button"
            >
              ›
            </button>
            <button
              aria-label=""
              class="react-calendar__navigation__arrow react-calendar__navigation__next2-button"
              type="button"
            >
              »
            </button>
          </div>
          <div
            class="react-calendar__viewContainer"
          >
            <div
              class="react-calendar__month-view"
            >
              <div
                style="display: flex; align-items: flex-end;"
              >
                <div
                  style="flex-grow: 1; width: 100%;"
                >
                  <div
                    class="react-calendar__month-view__weekdays"
                    style="display: flex; flex-wrap: nowrap;"
                  >
                    <div
                      class="react-calendar__month-view__weekdays__weekday"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    >
                      <abbr
                        aria-label="Monday"
                        title="Monday"
                      >
                        Mon
                      </abbr>
                    </div>
                    <div
                      class="react-calendar__month-view__weekdays__weekday"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    >
                      <abbr
                        aria-label="Tuesday"
                        title="Tuesday"
                      >
                        Tue
                      </abbr>
                    </div>
                    <div
                      class="react-calendar__month-view__weekdays__weekday"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    >
                      <abbr
                        aria-label="Wednesday"
                        title="Wednesday"
                      >
                        Wed
                      </abbr>
                    </div>
                    <div
                      class="react-calendar__month-view__weekdays__weekday"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    >
                      <abbr
                        aria-label="Thursday"
                        title="Thursday"
                      >
                        Thu
                      </abbr>
                    </div>
                    <div
                      class="react-calendar__month-view__weekdays__weekday"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    >
                      <abbr
                        aria-label="Friday"
                        title="Friday"
                      >
                        Fri
                      </abbr>
                    </div>
                    <div
                      class="react-calendar__month-view__weekdays__weekday react-calendar__month-view__weekdays__weekday--weekend"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    >
                      <abbr
                        aria-label="Saturday"
                        title="Saturday"
                      >
                        Sat
                      </abbr>
                    </div>
                    <div
                      class="react-calendar__month-view__weekdays__weekday react-calendar__month-view__weekdays__weekday--current react-calendar__month-view__weekdays__weekday--weekend"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                    >
                      <abbr
                        aria-label="Sunday"
                        title="Sunday"
                      >
                        Sun
                      </abbr>
                    </div>
                  </div>
                  <div
                    class="react-calendar__month-view__days"
                    style="display: flex; flex-wrap: wrap;"
                  >
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="December 30, 2024"
                      >
                        30
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="December 31, 2024"
                      >
                        31
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 1, 2025"
                      >
                        1
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 2, 2025"
                      >
                        2
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 3, 2025"
                      >
                        3
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 4, 2025"
                      >
                        4
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 5, 2025"
                      >
                        5
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 6, 2025"
                      >
                        6
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 7, 2025"
                      >
                        7
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 8, 2025"
                      >
                        8
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 9, 2025"
                      >
                        9
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 10, 2025"
                      >
                        10
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 11, 2025"
                      >
                        11
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 12, 2025"
                      >
                        12
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 13, 2025"
                      >
                        13
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 14, 2025"
                      >
                        14
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 15, 2025"
                      >
                        15
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 16, 2025"
                      >
                        16
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 17, 2025"
                      >
                        17
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 18, 2025"
                      >
                        18
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 19, 2025"
                      >
                        19
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 20, 2025"
                      >
                        20
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 21, 2025"
                      >
                        21
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 22, 2025"
                      >
                        22
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 23, 2025"
                      >
                        23
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 24, 2025"
                      >
                        24
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 25, 2025"
                      >
                        25
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__tile--now react-calendar__tile--active react-calendar__tile--range react-calendar__tile--rangeStart react-calendar__tile--rangeEnd react-calendar__tile--rangeBothEnds react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 26, 2025"
                      >
                        26
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 27, 2025"
                      >
                        27
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 28, 2025"
                      >
                        28
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 29, 2025"
                      >
                        29
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 30, 2025"
                      >
                        30
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="January 31, 2025"
                      >
                        31
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 1, 2025"
                      >
                        1
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                      style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden; margin-inline-end: 0;"
                      type="button"
                    >
                      <abbr
                        aria-label="February 2, 2025"
                      >
                        2
                      </abbr>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex items-center justify-between"
        >
          <h1
            class="text-xl font-semibold my-4"
          >
            Events
          </h1>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-calendar-days w-[20px] h-[20px]"
            data-icon="calendar-days"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M128 0c17.7 0 32 14.3 32 32l0 32 128 0 0-32c0-17.7 14.3-32 32-32s32 14.3 32 32l0 32 48 0c26.5 0 48 21.5 48 48l0 48L0 160l0-48C0 85.5 21.5 64 48 64l48 0 0-32c0-17.7 14.3-32 32-32zM0 192l448 0 0 272c0 26.5-21.5 48-48 48L48 512c-26.5 0-48-21.5-48-48L0 192zm64 80l0 32c0 8.8 7.2 16 16 16l32 0c8.8 0 16-7.2 16-16l0-32c0-8.8-7.2-16-16-16l-32 0c-8.8 0-16 7.2-16 16zm128 0l0 32c0 8.8 7.2 16 16 16l32 0c8.8 0 16-7.2 16-16l0-32c0-8.8-7.2-16-16-16l-32 0c-8.8 0-16 7.2-16 16zm144-16c-8.8 0-16 7.2-16 16l0 32c0 8.8 7.2 16 16 16l32 0c8.8 0 16-7.2 16-16l0-32c0-8.8-7.2-16-16-16l-32 0zM64 400l0 32c0 8.8 7.2 16 16 16l32 0c8.8 0 16-7.2 16-16l0-32c0-8.8-7.2-16-16-16l-32 0c-8.8 0-16 7.2-16 16zm144-16c-8.8 0-16 7.2-16 16l0 32c0 8.8 7.2 16 16 16l32 0c8.8 0 16-7.2 16-16l0-32c0-8.8-7.2-16-16-16l-32 0zm112 16l0 32c0 8.8 7.2 16 16 16l32 0c8.8 0 16-7.2 16-16l0-32c0-8.8-7.2-16-16-16l-32 0c-8.8 0-16 7.2-16 16z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="flex flex-col gap-4"
        >
          <div
            class="p-5 rounded-md border-2 border-gray-100 border-t-4 odd:border-t-preschoolPrimaryLight even:border-t-preschoolSecondaryLight"
          >
            <div
              class="flex items-center justify-between"
            >
              <h1
                class="font-semibold text-gray-600"
              >
                Lorem ipsum dolor
              </h1>
              <span
                class="text-preschoolPrimary-300 text-xs"
              >
                3:00PM - 7:00PM
              </span>
            </div>
            <p
              class="mt-2 text-gray-400 text-sm"
            >
              Lorem ipsum dolor sit amet
            </p>
          </div>
          <div
            class="p-5 rounded-md border-2 border-gray-100 border-t-4 odd:border-t-preschoolPrimaryLight even:border-t-preschoolSecondaryLight"
          >
            <div
              class="flex items-center justify-between"
            >
              <h1
                class="font-semibold text-gray-600"
              >
                Lorem ipsum dolor
              </h1>
              <span
                class="text-preschoolPrimary-300 text-xs"
              >
                12:00AM - 2:00AM
              </span>
            </div>
            <p
              class="mt-2 text-gray-400 text-sm"
            >
              Lorem ipsum dolor sit amet
            </p>
          </div>
          <div
            class="p-5 rounded-md border-2 border-gray-100 border-t-4 odd:border-t-preschoolPrimaryLight even:border-t-preschoolSecondaryLight"
          >
            <div
              class="flex items-center justify-between"
            >
              <h1
                class="font-semibold text-gray-600"
              >
                Lorem ipsum dolor
              </h1>
              <span
                class="text-preschoolPrimary-300 text-xs"
              >
                12:00PM - 2:00PM
              </span>
            </div>
            <p
              class="mt-2 text-gray-400 text-sm"
            >
              Lorem ipsum dolor sit amet
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
